2-mandatory/3-shopping-list.js

```diff
@@ -20,6 +20,21 @@ let pantry = {
 
 function createShoppingList(recipe) {
   // write code here
+
+  let missingIngredients = [];
+
+  // 2 lovely array methods which I learned here
+
+  const allPantryContents = Object.values(pantry).flat();
```

Comment:
This is a neat way to get the ingredients in the pantry into one list! Note that it wouldn't work if the pantry object had other properties too that didn't contain lists of ingredients. But if we know that it never will (and that might be quite reasonable here) then this is a good way to go.